### PR TITLE
case simple form

### PR DESCRIPTION
**Pull request by swilly22**
https://neo4j.com/docs/cypher-manual/current/syntax/expressions/#query-syntax-case

### CODE REVIEW

File: src/arithmetic/arithmetic_expression.c [hunk 1]
```diff
@@ -232,7 +232,32 @@ AR_ExpNode *AR_EXP_FromExpression(RecordMap *record_map, const cypher_astnode_t
 		const cypher_astnode_t *rhs_node = cypher_ast_binary_operator_get_argument2(expr);
 		op->op.children[1] = AR_EXP_FromExpression(record_map, rhs_node);
 		return op;
+	} else if(type == CYPHER_AST_CASE) {
+		/* Determin number of children expressions:
+		 * arg_count = value + 2*alternatives + default. */
+		unsigned int alternatives = cypher_ast_case_nalternatives(expr);
```
Comment (DvirDukhan) [inline src/arithmetic/arithmetic_expression.c]: Does libcypher parser hold different ast nodes or different number of expressions when in general case form?

File: src/arithmetic/arithmetic_expression.c [hunk 2]
```diff
@@ -978,6 +1004,35 @@ SIValue AR_TIMESTAMP(SIValue *argv, int argc) {
...
+		SIValue a = argv[i];
+		if(SIValue_Compare(v, a) == 0) {
+			// Return Result i.
+			return argv[i + 1];
+		}
+	}
+
+	/* Did not match against any Option
+	 * Return default. */
+	return d;
```
Comment (DvirDukhan) [inline src/arithmetic/arithmetic_expression.c]: Is d defaults to null?

File: src/ast/ast_mapping.c [hunk 3]
```diff
@@ -107,6 +107,9 @@ static void _AST_MapExpression(AST *ast, const cypher_astnode_t *expr) {
 		for(uint i = 0; i < nchildren; i ++) {
 			_AST_MapExpression(ast, cypher_ast_comparison_get_argument(expr, i));
 		}
+	} else if(type == CYPHER_AST_CASE) {
+		// TODO: implement...
```
Comment (DvirDukhan) [inline src/ast/ast_mapping.c]: Todo?

File: src/arithmetic/arithmetic_expression.c (hunk 2 above)
Comment (swilly22) [reply src/arithmetic/arithmetic_expression.c]: yep, if default is not specified it is defaults to NULL

File: src/arithmetic/arithmetic_expression.c (hunk 1 above)
Comment (swilly22) [reply src/arithmetic/arithmetic_expression.c]: in the simple case we've got:
1. value
2. a variable number of Alternatives (pairs)
3. Default value which might be null

In the generic form value is omitted.

**Comment by swilly22** [comment]
@DvirDukhan, I couldn't find any TCK feature files covering this feature, can you please verify that this is indeed the case? thanks!

File: src/arithmetic/arithmetic_expression.c (hunk 1 above)
Comment (DvirDukhan) [reply src/arithmetic/arithmetic_expression.c]: so as far as I understand, In the generic form calling `const cypher_astnode_t *expression = cypher_ast_case_get_expression(expr);` will yield NULL? how does the AR_CASEWHEN function can differ between an omitted value (generic) to a null evaluated expression?

Comment (swilly22) [reply src/arithmetic/arithmetic_expression.c]: by inspecting argc,
odd for generic, even for simple

Comment (DvirDukhan) [reply src/arithmetic/arithmetic_expression.c]: oh, missed this update :)

**Comment by DvirDukhan** [comment]
@swilly22 no TCK tests. we should either write our own and PR them or add as a flow test

**Pull request merged**